include/api.php

```diff
@@ -1524,41 +1524,53 @@ function api_search($type)
 	$a = get_app();
 	$user_info = api_get_user($a);
 
-	if (api_user() === false || $user_info === false) {
-		throw new ForbiddenException();
-	}
+	if (api_user() === false || $user_info === false) { throw new ForbiddenException(); }
 
-	$data = [];
-
-	if (empty($_REQUEST['q'])) {
-		throw new BadRequestException("q parameter is required.");
-	}
+	if (empty($_REQUEST['q']) && empty($_REQUEST['friendica_tag'])) { throw new BadRequestException("q or friendica_tag parameter is required."); }
 
+	$data = [];
+	$count = 15;
 	if (!empty($_REQUEST['rpp'])) {
 		$count = $_REQUEST['rpp'];
 	} elseif (!empty($_REQUEST['count'])) {
 		$count = $_REQUEST['count'];
-	} else {
-		$count = 15;
 	}
 
 	$since_id = defaults($_REQUEST, 'since_id', 0);
 	$max_id = defaults($_REQUEST, 'max_id', 0);
 	$page = (!empty($_REQUEST['page']) ? $_REQUEST['page'] - 1 : 0);
-
 	$start = $page * $count;
+	$params = ['order' => ['id' => true], 'limit' => [$start, $count]];
 
-	$condition = ["`gravity` IN (?, ?) AND `item`.`id` > ?
-		AND (`item`.`uid` = 0 OR (`item`.`uid` = ? AND NOT `item`.`global`))
-		AND `item`.`body` LIKE CONCAT('%',?,'%')",
-		GRAVITY_PARENT, GRAVITY_COMMENT, $since_id, api_user(), $_REQUEST['q']];
-
-	if ($max_id > 0) {
-		$condition[0] .= " AND `item`.`id` <= ?";
-		$condition[] = $max_id;
+	if (!empty($_REQUEST['q'])) {
+		$condition = [
+			"`id` > ? 
+			AND (`uid` = 0 OR (`uid` = ? AND NOT `global`))
+			AND `body` LIKE CONCAT('%',?,'%')",
+			$since_id, api_user(), $_REQUEST['q']];
+		if ($max_id > 0) {
+			$condition[0] .= " AND `id` <= ?";
+			$condition[] = $max_id;
+		}
+	} elseif (!empty($_REQUEST['friendica_tag'])) {
```

Comment:
Please replace this condition with a regular expression checking for a single hashtag in `q`.